lec07-pop-models.Rmd

```diff
@@ -1,4 +1,657 @@
 ---
-title: "Qualitative population models in 2 dimensions"
+title: "Numerically solving equations in R, population models in two dimensions"
+author: "Madeleine Bonsma-Fisher"
 ---
 
+> ### Learning objectives
+>
+> - Review differential equations and one-dimensional phase portraits
+> - Numerically solve differential equations in R using Euler's method and R's ODE solver
+> - Qualitatively analyze two-dimensional models by hand and in R
+
+> ### Lesson outline
+> 
+> Total lesson time: 2 hours
+> 
+> - 
+
+> ### Setup
+> 
+> - `install.packages('deSolve')`
+> - `install.packages('ggplot2')` (or `tidyverse`)
+> - `install.packages('dplyr')` (or `tidyverse`)
+
+## Recap and some tips
+
+   - You can include math in R notebooks using [LaTeX syntax](https://en.wikibooks.org/wiki/LaTeX/Mathematics). 
+
+     `$$\frac{dN}{dt} = r N (1 - N/K)$$` produces 
+     $$\frac{dN}{dt} = r N (1 - N/K)$$
+
+   - **Differential equation**: an equation that describes how a function of variables is related to
+   derivatives of those variables. 
+
+   - Drawing **phase portraits** in one dimension:
+
+    ![Logistic equation simulation](https://github.com/UofTCoders/rcourse/raw/master/logistic.gif)
+
+       - Fixed points: values of $N$ at which $dN/dt$, the rate of change of $N$, is $0$. 
+       To find fixed points, plot $dN/dt$ vs. $N$ and find the place(s) where it crosses the $x$ axis. 
+       - Stability: if you start at some $N$ close to the fixed point but not exactly on it, 
```

Comment:
Yeah, in a 1D system you could have "half-stable" where it's stable from one direction but unstable on the other side. 

In 2D systems, you can have the flow going towards a fixed point in one eigendirection and away in another. It's still technically unstable, but it has a special name: **saddle node**, because if you imagine the stability landscape as a hill, that fixed point is like a mountain pass.

Reply:
Makes sense, thanks for the explanation!